### ISSUE TITLE
Streaming (Sunshine or Steam Link) does not capture Steam UI

### ISSUE THREAD

**Issue by rcarmo** [ublue-os/bazzite#1276]
### Describe the bug

Right now, when streaming to another machine using Steam Link or `sunshine`, I cannot get the remote machine to see the Steam UI - it is simply not rendered in the stream.

That is, if I am playing a game:

![snapshot](https://github.com/ublue-os/bazzite/assets/392683/373440b6-3417-416b-8674-5f90f6a36655)

...and I hit the Steam button (or equivalent mapped to my remote controller), I will see this on the machine's physical display show this (captured with a PiKVM):

![snapshot-2](https://github.com/ublue-os/bazzite/assets/392683/352b56bd-ce2c-4525-9807-57b7d4dd49fd)

...and after pressing "down", the UI becomes translucent:

![snapshot-3](https://github.com/ublue-os/bazzite/assets/392683/816f3368-7096-4e52-a0dd-2034a9770fc0)

However, `sunshine`/Moonlight will display none of it. On the client machine will always see the game layer, without anything overlaid (although I can hear the UI audio):

![snapshot](https://github.com/ublue-os/bazzite/assets/392683/373440b6-3417-416b-8674-5f90f6a36655)

I have tried Steam Link settings, checked If `sunshine` is running with `setcap`, and also tried a number of different images (see #1260 to get an idea), but the issue happens on both `deck` and non-`deck` images, on GNOME or KDE.

Am curious to understand if this is a Wayland or AMD thing and if there are any fixes/workarounds.

### What did you expect to happen?

I expected the stream to match the physical display.

### Output of `rpm-ostree status`

```shell
State: idle
Deployments:
  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck-gnome:40-stable-20240611
                   Digest: sha256:26b268ec5de0c1618f0618855761b2da934614dfb060fd5d69ffed4fad14d2b9
                  Version: 40.20240611.0 (2024-06-11T16:59:32Z)
                     Diff: 460 removed, 193 added
          LayeredPackages: fio python3.11 python3.11-devel sunshine
                Initramfs: '"-I /etc/crypttab /etc/modprobe.d/amdgpu.conf /etc/modprobe.d/deck-blacklist.conf"' 

● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck:40-stable-20240611
                   Digest: sha256:6c59a228186bdbd3417580129702a9337c2edadac0206b0d4773df53dca58413
                  Version: 40.20240611.0 (2024-06-11T17:01:53Z)
          LayeredPackages: fio python3.11 python3.11-devel sunshine
                Initramfs: '"-I /etc/crypttab /etc/modprobe.d/amdgpu.conf /etc/modprobe.d/deck-blacklist.conf"' 

  ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck-gnome:40-stable-20240611
                   Digest: sha256:26b268ec5de0c1618f0618855761b2da934614dfb060fd5d69ffed4fad14d2b9
                  Version: 40.20240611.0 (2024-06-11T16:59:32Z)
          LayeredPackages: fio python3.11 python3.11-devel sunshine
                Initramfs: '"-I /etc/crypttab /etc/modprobe.d/amdgpu.conf /etc/modprobe.d/deck-blacklist.conf"'
```

### Hardware

```neofetch
                                            me@bazzite
     %%%%%%====%%%%%%%%%%                
   %%%%%%%%    %%%%%%%%%%%%%%             󱋩  bazzite-deck:stable 
  %%%%%%%%%    %%%%%%%%%%%%%%%%           󰣛  Fedora Linux 40 (Kinoite) x86_64
  %%%%%%%%%    %%%%%%%%%%%%%%%###           Linux 6.8.12-301.fsync.fc40.x86_64
  %%%%%%%%%    %%%%%%%%%%%%%######        󰅐  2 hours, 30 mins
  ==                  =======######       󰔠  Spawned on Feb 27 2024
  ==                  =========#####     
  %%%%%%%%%    %%%%%%%####======#####     󰾰  HCAR6000-MI2
  %%%%%%%%%    %%%%%#######=====#####     󰻠  AMD Ryzen 7 7840HS w/ Radeon 780M Graphics (16) @ 6.08 GHz
  %%%%%%%%%    %%%#########=====#####     󰍛  AMD Radeon 780M @ 0.80 GHz [Integrated]
  %%%%%%%%%    %%##########=====#####       5.15 GiB / 30.60 GiB (17%)
  %%%%%%%%%====###########=====######       384.69 GiB / 475.35 GiB (81%) - btrfs [Read-only]
   %%%%%%%%====#########======######      󰍹  1920x1080 @ 50Hz [External]
    %%%%%%%=====#####========######       󰖺  PiKVM Composite KVM Device
     %%%%###===============#######        󰖺  Mouse passthrough
      %#######==========#########        
        #######################             bash 5.2.26
          ###################               /dev/pts/1
              ###########                 󰏖  2717 (rpm), 79 (flatpak-system), 41 (flatpak-user)

                                           ● ● ● ● ● ● ● ●
```

**Comment by HikariKnight** [comment]
Don't think this is something we can fix as i vaguely remember this also being a problem on SteamOS

**Comment by dm-zharov** [comment]
> Don't think this is something we can fix as i vaguely remember this also being a problem on SteamOS

I haven’t encountered this issue on SteamOS 3.6.6

**Comment by rcarmo** [comment]
I've been investigating this with a few friends who have SteamDecks and as it turns out, this week I confirmed this works on SteamOS (with SteamLink). And I have a recollection of it working in earlier releases in both Steam Link and `sunshine`, although it was almost a year back and in another device.

**Comment by Darksk0pe** [comment]
Same problem  hope you can fix it :)

**Comment by thekk1** [comment]
I have this issue too on my UM790pro.
Additional the mouse cursor is not rendered, too. So mouse/keyboard games are impossible at remote play.

This works fine with ChimeraOS and with an SteamDeck with original SW.

This tells me that this is a real issue in bazzite.

**Comment by thekk1** [comment]
I have done more research and it turns out that there are more issues with remote play than the two above.
Here is a complete list:

- No Steam overlay while remote play
    - Works with SteamOS, ChimeraOS and clouddeck.app
    - Affects any client (SteamLink HW, Android, Linux, Windows)
- No mouse cursor if mouse is connected with the client device
    - Only testes with SteamLink HW
- Switching to desktop mode breaks BPM and so Remote Play, too
    - Workaround is to restart steam in desktop mode or disable BPM in remote device
- In desktop mode the steam overlay works but the background is not visible. There is only the menu visible and centered like "full screen"
- Remote play in desktop mode breaks the mouse cursor for the gaming|server device if there is a mouse connected at the remote device. There is no cursor visible until you restart.

**Comment by kasper-madsen** [comment]
I sort of have this but also however using mouse mode via Xbox controller brings the overlay up or move the mouse via my iPad also fixes this but when the mouse hides so does the overlay

**Comment by TeleBooth** [comment]
I’m running into this issue the same way that @kasper-madsen is describing it. I have tried this on Nobara 40, the latest Bazzite, and even the latest ChimeraOS. All the same. 

I have a feeling that something changed around May/June that has affected all of these builds. I can try backdating to an older version of these systems to see if it works.

**Comment by Zaunei** [comment]
When I activate the performance overlay on level 2, I suddenly see the Steam overlay within the Sunshine stream, but the performance overlay is then only displayed on the real monitor. 


Otherwise, I also have issues when I set a game to a lower resolution (in the game itself) than the normal resolution used in Steam:
![Screenshot 2024-09-29 at 10 06 24@2x](https://github.com/user-attachments/assets/246a61d2-4e2c-4d6a-ab9c-eef0db6dfa54)

As soon as the Steam overlay is displayed, it scales correctly.

![Screenshot 2024-09-29 at 10 06 04@2x](https://github.com/user-attachments/assets/b1272a72-87f8-4c38-9d39-17b86ed2cb50)

**Comment by WeslyG** [comment]
I tested moonlight on different clients, and found in the native client for mac os that when you press the Esc button, the window turns black, but the steam menus that were previously unavailable are transformed. But only they (left and right). It's funny, and I haven't found a way to get back to regular steam other than restarting the moonlight session.

**Comment by floppyFK** [comment]
I have the same problem: No Steam overlay is visible in a streamed session (sunshine/moonlight as well as steam link). On the Host PC however, I can see the overlay. I use the bazzite-deck-stable.iso (KDE) image on my desktop PC with an AMD 7900XTX.

Now I have tested also the bazzite-stable.iso (KDE), where i enabled the big picture on startup of Steam and voila: The Steam overlay appears now also in the stream.
The problem seems to be related to the deck-specific GUI of Steam.

**Comment by nlflint** [comment]
Btw, I'm seeing this bug in Arch too, so I don't think it's a Bazzite bug. I'm trying to repo it on the lastest official release of Sunshine (v0.23.1), as Bazzite is on an unstable Sunshine version, and only the unstable Sunshine packages work from the AUR. I'm trying to build from the official repo on tag `v0.23.1`.

**Comment by nlflint** [comment]
>  will report back later at what I find.

For the life of me, I cannot build sunshine v0.23.1 on Arch, so many deprecated libs, especially boost 1.83. So instead, I tested on a ChimeraOS v45-1 machine with sunshine v0.23.1, and it does NOT exhibit this bug.

Then I upgraded my Chimera box to Sunshine to 'v2024.1102.190004.r0.g39bab45-1`  (AUR package sunshine-git), and Sunshine still does NOT exhibit this bug. Therefore, I don't think the bug is directly related to Sunshine version. As this Sunshine version exhibits the bug on my Arch box.

Next, I'm going to try to downgrade the kernel on my Arch box to match ChimeraOS 45-1 (v6.6.10), and see if the bug repros on Arch.

**Comment by nlflint** [comment]
I tested back to kernels 6.5.0 and 6.6.64 (LTS), and the bug still exists, so it's probably not kernel related either. So strange that it doesn't repro on my Chimera v45-1 box.

**Comment by scottsweb** [comment]
@nlflint It happens on both Sunshine and Steam Link, so I have ruled out Sunshine as the culprit for now too.

If I remember correctly, you can get it work if you switch to desktop in Bazzite and then stream a game from there. This skips the `gamescope`  specific session, but it comes at the cost of a few nice things like scaling controls etc.

There seems to be lots of people with similar issues:

* https://www.reddit.com/r/SteamDeck/comments/185b57d/steam_and_button_not_working_during_remote_play/
* https://steamcommunity.com/groups/homestream/discussions/0/4353366316815125003/
* https://www.reddit.com/r/Steam/comments/1dnyhb4/remote_play_overlay_not_showing/
* https://www.reddit.com/r/SteamDeck/comments/1899g34/remote_play_overlay/

**Comment by nlflint** [comment]
Between my two machines (one with the big one without), they share the same
Steam steam version (1733265492, from BigPicture -> Settings -> System) so
I don’t think the cause is steam either.

**Comment by nlflint** [comment]
Update: I was able to make the bug go away on Arch, but I had to take some extreme measures to do it:

1. Installed fresh Arch with all packages locked to official package archives from Feb 2nd 2024. Almost year old packages at this point.
2. Installed the various gamescope session packages from the AUR, but resetting the package build repos to their commits on Feb 2nd.
3. Installed the exact same linux kernel as ChimeraOS 45-1 (v6.6.10-chos1-chimeraos-2). They've got custom patches in there...

I'm going bisect this stuff until I can narrow down the problem. Probably gonna take awhile. However, it does seem kernel related, as a regular non-patched kernel brings the bug back.

**Comment by nlflint** [comment]
Update: 

On my test system, I've bisected the bug to two components, it takes the right combination of versions for the bug to appear/disapper:
1. Gamescope
2. Chimera kernel patches

For gamescope, this commit is where the bug first shows up: https://github.com/ValveSoftware/gamescope/commit/04ad0c9be89cfa2a4152908fd2bfe16743591757 titled: `drm: Update AMD color management for Linux 6.8`. This commit was reverted for several releases after it was introduced. Finally, it was permanently present in release `v3.14.8` and later. The latest release where the bug does NOT reproduce is `v3.14.7`.

Technically my bisect was done against ChimeraOS's fork of Gamescope, where this screenshot comes from. Chimera keeps it's fork up to date, build from these commits will exactly match Valves repo. Note, I marked commits with branch names `working` and `broken` to indicated which commits contain the undesired behavior:
![Image](https://github.com/user-attachments/assets/b608d4ca-96e4-481a-b58b-6e52002dcf10)


For the kernel, I'm using ChimeraOS's kernel. My bisect located two boundaries. Essentially, the bug appears in version `v6.6.4-chos2-2` and earlier, goes away from `v6.6.4-chose3-1` to v6.6.17-chos2-1`. Finally the bug reappears in all their `v6.8.x` builds and later. See screenshot with my branches marked `bugged` vs `works` to see what I tested. For this, I haven't nailed down the actual kernel commits. I looked into it and the diff is hard to make sense of. I plan to better bisect this as I have time.
![Image](https://github.com/user-attachments/assets/ae6452e1-30f9-411b-b5d1-407f6c0ff4cd)


Bottom line, the bug GOES AWAY when these two conditions are met:
1. When your `Gamescope` or `gamescope-plus` version is `v3.14.7` or earlier.
2. Your ChimeraOS's kernel is `v6.6.4-chos3-1` to `v6.6.17-chos2-1`

Any combination outside those versions will contain this sunshine overlay bug.

**Comment by speakradmin** [comment]
Amazing work @nlflint !

Really hoping that this infuriating bug can be fixed soon

**Comment by chunky-games** [comment]
@nlflint I have found a weird glitch with this as well. if all settings in steams displays are left to default, then you enter a game and change the in-game resolution to below the monitor's base resolution, the steam UI will appear and work as normal. but this work around is extremely settings based, for example. in quick settings when you press Xbox button + A (or whatever controller you play with)  your per game profile needs to be switched on and the scaling filter has to be switched off of linear, either to pixel, fsr, or nis. It's extremely odd but it works for some reason.

And just for reference, I'm on the regular bazzite-deck release.

**Comment by chunky-games** [comment]
@chunky-games also in the steam quick settings you need to have disable frame limit turned on. or else the steam UI will not work also

**Comment by bugs181** [comment]
> I sort of have this also, but however using mouse mode via Xbox controller brings the overlay up or move the mouse via my iPad also fixes this but when the mouse hides so does the overlay

You are a life saver! This workaround works for me too. Had I known this, I would have been enjoying decky loader far more for a couple weeks now.

**Comment by nlflint** [comment]
@bugs181 @chunky-games 

None of those workarounds made a difference on my machine (I'm on arch, so maybe because I have slightly different package versions). I tried playing around with more settings and nothing fixed it but....

**Enabling HDR** fixes it! (for the most part). I have a new overlay issue with one of my games, but the overlays work as expected in all the other games I tested. HDR looks great on my livingroom TV w/AppleTV4k. However, it's not a permanent fix as I have a client w/o an HDR monitor and the colors are messed up when in HDR.

**Comment by bugs181** [comment]
> [@bugs181](https://github.com/bugs181) [@chunky-games](https://github.com/chunky-games)
> 
> None of those workarounds made a difference on my machine (I'm on arch, so maybe because I have slightly different package versions). I tried playing around with more Steam display settings, and QAM settings, and nothing fixed it but....
> 
> **Enabling HDR** fixes it! (for the most part). I have a new overlay issue with one of my games, but the overlays work as expected in all the other games I tested. HDR looks great on my livingroom TV w/AppleTV4k. However, it's not a permanent fix as I have a client w/o an HDR monitor and the colors are messed up when in HDR.
> 
> This is with the latest packages and kernel as of today on Arch.

I'll try to give HDR setting a go. Unfortunately, I don't have any HDR capable displays yet. Depending how bad the colors are messed up, it might be a temporary workaround worth using if it works.

**Comment by dazzla76** [comment]
I started getting this issue when I switched from a rx570 to rx7800xt a couple of weeks ago.

**Comment by neowave2048** [comment]
I'm also having this issue with my AMD cards (RX 7900 GRE & RX 7600M XT.) I tried @nlflint's above suggestion and enabled HDR on my supported TV and tablet, but to no avail. The only way to make the overlay appear is to toggle on mouse emulation mode in Moonlight by holding the start button on my controller and moving the left analog stick around. This'll make the cursor and overlay appear.

**Comment by dazzla76** [comment]
That’s an interesting workaround. I’ll give it a go as well.

**Comment by TariqMK** [comment]
I arrived at this thread after spending a lot of time troubleshooting not knowing what the culprit was. Like you all, I am experiencing this exact issue.

Remotely using Moonlight, I dont get the overlay showing.

> I'm also having this issue with my AMD cards (RX 7900 GRE & RX 7600M XT.) I tried [@nlflint](https://github.com/nlflint)'s above suggestion and enabled HDR on my supported TV and tablet, but to no avail. The only way to make the overlay appear is to press the guide button (e.g. Xbox and Playstation middle button,) and toggle on mouse emulation mode in Moonlight by holding down the start button on my controller for 2-3 seconds and releasing. Then I move the left analog stick around, and this will make the cursor and overlay appear.

I can confirm this workaround works for me, so for some reason the overlay does not display normally when using a controller over Moonlight.

Here's hoping a fix for this one comes soon!

**Comment by dazzla76** [comment]
I can confirm this works for me on my Odin 2 device. A little bit fiddly but at least it’s something for now.

**Comment by Geofferey** [comment]
Seems this is not only Linux related. I tested steam on Windows with big picture enabled and attempted to bring up the overlay while connected via Moonlight... SAME exact issue as Bazzite.

**Comment by baltimoresports** [comment]
I confirmed the issue is present in SteamLink as well as Moonlight.

**Comment by baltimoresports** [comment]
I opened a bug issue over at the Sunshine Github:
https://github.com/LizardByte/Sunshine/issues/3860

**Comment by Gilibenzio** [comment]
I am having the same issues in Bazzite and I have also an all AMD PC (Ryzen 5 5600, RX 6600).

**Comment by nlflint** [comment]
Another workaround reported by someone: https://github.com/LizardByte/Sunshine/issues/3860#issuecomment-2954267416

> I stumbled across a workaround for this. Turn on developer mode under system then go to developer mode and turn on "Force Composite". However the only caveat is you need to turn HDR off otherwise it will look washed out. It works with Bazzite and and Cachyos handheld. It should work with other Steam OS like distros as well.

**Comment by Borotes** [comment]
I have this Problem with Sunshine -> Moonlight, or Steam Remote Play -> Windows Steam Remote Play on my Bazzite Machine, and from my Steam Deck (Official Steam OS) ->Windows Steam Remote Play. On the Homescreen of Gamescope it is working, but not when starting a game.

rpm-ostree status:
State: idle
Deployments:
● ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck:stable
Digest: sha256:308783769ddf7450b76837670cfb4eef915f42b04fca85d6c18efe67e289d78c
Version: 42.20250708.1 (2025-07-09T00:37:02Z)

ostree-image-signed:docker://ghcr.io/ublue-os/bazzite-deck:stable
Digest: sha256:e7efe7a4715cfa4f74573954245019db49942ae13fd33487a77b43622b9ba435
Version: 42.20250708 (2025-07-08T05:41:28Z)

My Bazzite Machine has an RX 6700 XT.

**Comment by nlflint** [comment]
> Another workaround reported by someone: https://github.com/LizardByte/Sunshine/issues/3860#issuecomment-2954267416

Replying to my last post, I finally got around to trying that workaround and it solves this bug for me on Arch with latest versions of everything.

**Comment by bernhardberger** [comment]
> > Another workaround reported by someone: https://github.com/LizardByte/Sunshine/issues/3860#issuecomment-2954267416
> 
> Replying to my last post, I finally got around to trying that workaround and it solves this bug for me on Arch with latest versions of everything.
> 

For me forcing composite introduces wrong colors (too warm), but at least the overlay appears.

I have a feeling this might be gamescope+AMD related though as this appears across distros (also present on CachyOS with gamescope-session).

This appears on my AMD RX9070 machine, but isn't present on my Nvidia 4090 machine (running CachyOS, also with gamescope-session)

**Comment by Borotes** [comment]
Does the workaround still work for you all? I had it enabled on both of my devices, and it doesn't work on my Steam Deck anymore.

Everything is up to date there. I didn't do any special updates in the past weeks, and retoggling the option doesn't work.

**Comment by bernhardberger** [comment]
Yes, it does. However something is off with color management on AMD, or at least RDNA4.

- Force Composite off, HDR off -> too warm, reddish color
- Force Composite off, HDR off -> looks fine
- Force Composite off/on (doesn't matter), HDR on -> extremely cold color, very blue-ish

None of those happen on my nvidia rig, composite on/off, HDR on/off - all combinations look fine.

**Comment by youralmight** [comment]
Running bazzite on pve, accessing it from steamlink/sunshine, both seeing this bug.

**Comment by rbmaster** [comment]
Have this issue both with a 6700XT and 9070XT in game mode.
Force composite on or off makes no difference.

The steam overlay, or other overlays like reshare, result in a black screen or a frame from time to time (long periods, like 10-30 seconds).

**Comment by Borotes** [comment]
> Does the workaround still work for you all? I had it enabled on both of my devices, and it doesn't work on my Steam Deck anymore.
> 
> Everything is up to date there. I didn't do any special updates in the past weeks, and retoggling the option doesn't work.

Quick follow-up from my side: Steam Link is broken again (the workaround does not work anymore) with the latest Steam updates (both Bazzite and official SteamOS), but the overlay still works with Moonlight. The last Sunshine release was very unstable for me; that is why I had to switch temporary.
